### PR TITLE
fix: Warn: A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event.

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Fix warning:

> Warn: A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event. (svelte)

# Note about a11y

I just catched `Enter`, don't know if we should also catch `space bar` @mstrasinskis ?

# Changes

- `div` that behaves have button should:
   - `role=button`
   - `tabindex=0` or `-1`
   - `on:keypress`

### CODE REVIEW

File: src/lib/components/Collapsible.svelte [hunk 1]
```diff
@@ -129,6 +130,12 @@
       align-items: center;
       justify-content: flex-start;
     }
+
+    outline: none;
+
+    &:focus {
+      filter: contrast(1.25);
```
Comment (mstrasinskis) [inline src/lib/components/Collapsible.svelte:137]: Out of curiosity, is it noticeable on text?

File: src/lib/utils/keyboard.utils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const onEnter = ({
+  $event: { key },
+  callback,
+}: {
+  $event: KeyboardEvent;
+  callback: () => void;
+}) => {
+  if ("Enter" !== key) {
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/lib/utils/keyboard.utils.ts]: Following?

```
export const handleKeyPress = ({
  $event: { code },
  callback,
}: {
  $event: KeyboardEvent;
  callback: () => void;
}) => {
  if (!["Enter", "Space"].includes(code)) {
    return;
  }

  callback();
};
```

File: src/lib/components/Collapsible.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/lib/components/Collapsible.svelte:137]: Not really but, did not had good idea. `text-decoration: underline;` or better idea?

Comment (mstrasinskis) [reply src/lib/components/Collapsible.svelte:138]: Thanks, I think it's good as it is 👍

Maybe something for later to make tab-navigation focus more prominent e.g.
```
&:focus-visible {
  outline: var(--padding-0_25x) solid var(--warning-emphasis);
  outline-offset: calc(-1*var(--padding-0_25x));
}
```
image

File: src/lib/utils/keyboard.utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply src/lib/utils/keyboard.utils.ts]: Looks good!

File: src/lib/components/Collapsible.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/lib/components/Collapsible.svelte:138]: That's a great idea! If you want to give it a try later, go for it

**Pull request merged**